Kernel/FileSystem/BlockBasedFileSystem.cpp

```diff
@@ -99,8 +100,15 @@ class DiskCache {
             callback(entry);
     }
 
+    void clean_before_clear()
+    {
+        m_dirty_list.clear();
+        m_clean_list.clear();
+        m_hash.clear();
+    }
+
 private:
-    BlockBasedFileSystem& m_fs;
+    NonnullRefPtr<BlockBasedFileSystem> m_fs;
```

Comment:
There are several other mutable members here. should we make m_fs mutable as well to avoid the const_cast on line 75?